imgui-core/src/main/kotlin/imgui/internal/api/docking.kt

```diff
@@ -341,26 +341,18 @@ interface docking {
         val payloadWindow = payload.data as Window
         acceptDragDropPayload(IMGUI_PAYLOAD_TYPE_WINDOW, DragDropFlag.AcceptBeforeDelivery or DragDropFlag.AcceptNoDrawDefaultRect)?.let {
             // Select target node
-            var node: DockNode? = null
-            var allowNullTargetNode = false
-            window.dockNodeAsHost.let { host ->
-                if (host != null)
-                    node = dockNodeTreeFindNodeByPos(host, io.mousePos)
-                else window.dockNode.let {
-                    if (it != null) // && window->DockIsActive)
-                        node = it
-                    else
-                        allowNullTargetNode = true // Dock into a regular window
-                }
-            }
+            val node: DockNode? = window.dockNodeAsHost?.let {
+                dockNodeTreeFindNodeByPos(window.dockNodeAsHost!!, g.io.mousePos)
+            } ?: window.dockNode
+
             val explicitTargetRect = node?.let { n -> n.tabBar?.let { t -> if (!n.isHiddenTabBar && !n.isNoTabBar) Rect(t.barRect) else null } }
                     ?: Rect(window.pos, window.pos + Vec2(window.size.x, frameHeight))
             val isExplicitTarget = io.configDockingWithShift || isMouseHoveringRect(explicitTargetRect.min, explicitTargetRect.max)
 
             // Preview docking request and find out split direction/ratio
             //const bool do_preview = true;     // Ignore testing for payload->IsPreview() which removes one frame of delay, but breaks overlapping drop targets within the same window.
             val doPreview = payload.preview || payload.delivery
-            if (doPreview && (node != null || allowNullTargetNode)) {
```

Comment:
I don't get why you can skip that condition (especially `allowNullTargetNode`)

Reply:
Oops, I made the wrong assumption of thinking that `allowNullTargetNode` is only `true` if node is null, I'll revert this.

Reply:
If you want to keep the nice compact `?:` chain, you may consider appending `?: null.also { allowNullTargetNode = true }